### ISSUE TITLE
Can't delete an HttpApi event

### ISSUE THREAD

**Issue by maksimleanovich** [aws/serverless-application-model#1939]
### Description:
I'm working on a SAM application, and as part of it I have defined an `AWS::Serverless::HttpApi` and a `AWS::Serverless::Function` which is triggered by an HttpApi event. The creation process went fine, but later on I realized that I don't need this function and tried to remove it from the template. Unfortunately, I wasn't able to apply the change due to the following error:

```
Unable to find Route by key GET /watchlist within the provided RouteSettings (Service: AmazonApiGatewayV2; Status Code: 404; Error Code: NotFoundException; Request ID: b53fe8e1-03c2-4b88-8748-04c6a8a902d7; Proxy: null)
```

I first encountered this error when simply trying to change the route from `GET` to `POST` for another function, but at that time I simply deleted the entire stack and deployed it from scratch with the desired definition. This time, however, I can't go that route, as I already have some valuable data in Dynamo tables, which are part of the stack and would get deleted too. I need to migrate it somehow. What would be the best workaround for this? Thank you for any help!

### Steps to reproduce:
1. Define a SAM template like this:
```
...
  WatchlistReportHandler:
    Type: AWS::Serverless::Function
    Properties:
      FunctionName: !Sub "${stackPrefix}-watchlist-report-handler"
      Handler: slack_callback.lambda_handler
      Events:
        ExplicitApi:
          Type: HttpApi
          Properties:
            ApiId: !Ref HttpApi
            Method: GET
            Path: /watchlist
            TimeoutInMillis: 15000
            PayloadFormatVersion: "2.0"
            RouteSettings:
              ThrottlingBurstLimit: 600

  HttpApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      StageName: !Sub "${stackPrefix}-slack-api"
      AccessLogSettings:
        DestinationArn: !GetAtt AccessLogs.Arn
        Format: $context.requestId
      DefaultRouteSettings:
        ThrottlingBurstLimit: 200

  AccessLogs:
    Type: AWS::Logs::LogGroup

...
```

2. Deploy the stack
3. Try removing the function, just the event or event changing `GET` to `POST` in the route definition
4. Apply the change

### Observed result:
Stack update fails, saying "Unable to find Route by key"

```
...
UPDATE_IN_PROGRESS       AWS::ApiGatewayV2::Api   HttpApi                  -
UPDATE_COMPLETE          AWS::ApiGatewayV2::Api   HttpApi                  -
UPDATE_IN_PROGRESS       AWS::ApiGatewayV2::Sta   HttpApiStage             -
                         ge
UPDATE_FAILED            AWS::ApiGatewayV2::Sta   HttpApiStage             Unable to find Route
                         ge                                                by key GET /watchlist
                                                                           within the provided
                                                                           RouteSettings
                                                                           (Service:
                                                                           AmazonApiGatewayV2;
                                                                           Status Code: 404;
                                                                           Error Code:
                                                                           NotFoundException;
                                                                           Request ID: b53fe8e1-0
                                                                           3c2-4b88-8748-04c6a8a9
                                                                           02d7; Proxy: null)
UPDATE_ROLLBACK_IN_PRO   AWS::CloudFormation::S   github-signals-staging   The following
GRESS                    tack                                              resource(s) failed to
                                                                           update:
                                                                           [HttpApiStage].
...
```

### Expected result:
Successful deletion/change of the Function

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Mac OS
2. If using SAM CLI, `sam --version`: 1.18.2
3. AWS region: us-west-1

`Add --debug flag to any SAM CLI commands you are running`

**Comment by hawflau** [comment]
Hey @maksimleanovich I managed to reproduce your issue. 

The error you got was because of the RouteSetting of the API Stage. When I tested with a pure CloudFormation template with `RouteSettings` in a `AWS::ApiGatewayV2::Stage` stage, the error still exists. I'll file an internal ticket to API Gateway Team to inquire about this.

Meanwhile, there is a workaround - first remove `RouteSettings` in your `HttpApi` event, deploy it. Then make the change in your event and add back `RouteSettings`, then deploy again.
